OnvifSharp/Discovery/Models/DiscoveryDevice.cs

```diff
@@ -1,55 +1,11 @@
-﻿using System.Collections.Generic;
-using System.Linq;
-using System.Net;
-
-namespace OnvifSharp.Discovery.Models
+﻿namespace OnvifSharp.Discovery.Models
 {
 	public class DiscoveryDevice
 	{
-		public IEnumerable<string> Types { get; internal set; }
-		public IEnumerable<string> XAdresses { get; internal set; }
-		public string Model { get; internal set; }
-		public string Name { get; internal set; }
-		public IPAddress Address { get; internal set; }
-
-		public override bool Equals (object obj)
-		{
-			var o = obj as DiscoveryDevice;
-			if (o == null) {
-				return false;
-			}
-			if (o.Model != Model || o.Name != Name || o.Address != Address
-				|| o.Types.Count() != Types.Count() || o.XAdresses.Count() != XAdresses.Count()) {
-				return false;
-			}
-			for (var i = 0; i < Types.Count(); i++) {
-				if (!Types.ElementAt(i).Equals (o.Types.ElementAt(i))) {
-					return false;
-				}
-			}
-			for (var i = 0; i < XAdresses.Count (); i++) {
-				if (!XAdresses.ElementAt (i).Equals (o.XAdresses.ElementAt (i))) {
-					return false;
-				}
-			}
-			return true;
-		}
-
-		public override int GetHashCode ()
-		{
-			var hash = 1;
-			if (Types != null) {
-				foreach (var type in Types) {
-					hash += type.GetHashCode ();
-				}
-			}
-			if (XAdresses != null) {
-				foreach (var address in XAdresses) {
-					hash += address.GetHashCode ();
-				}
-			}
-			hash += (Model?.GetHashCode () + Name?.GetHashCode () + Address?.GetHashCode ()) ?? 0;
-			return hash;
-		}
+		public string Model { get; set; }
```

Comment:
This new model missed the overriden equals and get hash code, this is necessary for the DiscoveryService to work as expected, to avoid adding to the observable collection the same device since it will be from different instances, but could contain the same values.